src/patterns/kinesis-lambda.ts

```diff
@@ -49,29 +44,29 @@ export interface CrossAccountKinesisStream {
  * MyCloudFormedKinesisStream:
  *   Type: AWS::Kinesis::Stream
  * ```
- * Inherit the Kinesis stream (rather than creating a new one) using:
- * ```typescript
- * existingKinesisStream: { existingLogicalId: "MyCloudFormedKinesisStream" }
- * ```
  *
  * Alternatively, reference a Kinesis stream which belongs to another stack or pattern in the same aws account using:
  * ```typescript
- * existingKinesisStream: { externalKinesisStreamName: "KinesisStreamFromAnotherStack" }
+ * existingKinesisStream: { streamArn: "existingKinesisStreamArn" }
  * ```
  *
  *  Alternatively, reference a Kinesis stream which belongs to another stack or pattern in a different aws account using:
  * ```typescript
  * existingKinesisStream: {
- *   crossAccountKinesisStream: {
- *      roleArn: "CrossAccountKinesisStreamRoleArn",
- *      streamArn: "CrossAccountKinesisStreamArn",
- *   }
+ *    streamArn: "existingKinesisStreamArn",
+ *    roleArn: "CrossAccountKinesisStreamRoleArn",
  * }
  * ```
  */
 export interface ExistingKinesisStream extends GuMigratingResource {
-  externalKinesisStreamName?: string;
-  crossAccountKinesisStream?: CrossAccountKinesisStream;
+  /**
+   * kinesis stream Arn that is owned by a different stack
+   */
+  streamArn: string;
```

Comment:
Following @jacobwinch 's [comments](https://github.com/guardian/cdk/pull/1264#discussion_r872379592), could this be a [`GuArnParameter`](https://github.com/guardian/cdk/blob/a8f7692928ba7a1691aabb8260ef3777adcae16c/src/constructs/core/parameters/base.ts#L35-L43) type? We'll then be using the type system to avoid adding private/secret information into VCS.

Parameter's have a [`valueAsString` function](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.CfnParameter.html#valueasstring) that can be used to read the value when needed.

I think similar applies to `roleArn` below too?

Reply:
Sorry @akash1810 yesterday when @jacobwinch and I were trying to deploy the lambda that was using the cdk changes for kinesis lambda, we realised that  there's a limitation in AWS that doesn't allow a lambda to be triggered by a cross account kinesis stream 😢 

So we have to put the lambda in the same account as the kinesis stream meaning all of these changes we were making on the cdk project are not needed at all. But thanks a lot for looking into the changes and helping out 🙏